### PR TITLE
Up list

### PR DESCRIPTION
**Pull request by pingzh**
Fix some bugs and add a feature select a up in the up list, to up detail.

### CODE REVIEW

File: Groupon UP/Groupon UP/UPListViewController.swift [hunk 1]
```diff
@@ -76,7 +77,14 @@ class UPListViewController: BaseViewController {
     func initData() {
         grouponUps = []
         tableView.reloadData()
-        loadInvitingUps()
+        switch selectedSegmentIndex {
+        case 0:
```
Comment (roboxue) [inline Groupon UP/Groupon UP/UPListViewController.swift]: You can just remove this clause, as it has been covered in Default

File: Groupon UP/Groupon UP/UPListViewController.swift [hunk 2]
```diff
@@ -201,9 +192,7 @@ extension UPListViewController {
 extension UPListViewController {
     func refreshView() {
         refreshController.beginRefreshing()
-        grouponUps = []
         initData()
-        tableView.reloadData()
```
Comment (roboxue) [inline Groupon UP/Groupon UP/UPListViewController.swift:206]: better to call endRefreshing in the callback of initData....

File: Groupon UP/Groupon UP/UpListTableViewCell.swift [hunk 3]
```diff
@@ -77,21 +77,36 @@ class UpListTableViewCell: UITableViewCell {
...
+                }
             }
         }
+        else {
+            dealImage.af_setImageWithURL(NSURL(string: up.associatedDeal!.dealImages.sidebarImageUrl)!)
+            dealNameLabel.text = up.associatedDeal!.title
+        }
+        
+        let upDate = up.date.description
+        let dataRange = upDate.startIndex..<upDate.startIndex.advancedBy(10)
```
Comment (roboxue) [inline Groupon UP/Groupon UP/UpListTableViewCell.swift]: Use a `NSDateFormatter`. `.DateStyle` should be able to help you.

**Comment by roboxue** [comment]
Minor styling issues. Can try to fix them if you still have time, or I'll fix them when I do the release tonight

File: Groupon UP/Groupon UP/UPListViewController.swift (hunk 1 above)
Comment (pingzh) [inline Groupon UP/Groupon UP/UPListViewController.swift]: Fixed.

File: Groupon UP/Groupon UP/UPListViewController.swift (hunk 2 above)
Comment (pingzh) [inline Groupon UP/Groupon UP/UPListViewController.swift:206]: Fixed.

File: Groupon UP/Groupon UP/UpListTableViewCell.swift (hunk 3 above)
Comment (pingzh) [inline Groupon UP/Groupon UP/UpListTableViewCell.swift]: God, genius! Fixed

**Pull request merged**